packages/client/src/searchStreams.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable padding-line-between-statements */
+import { EthereumAddress } from 'streamr-client-protocol'
+import { StreamPermission } from './Stream'
+import { ChainPermissions, PUBLIC_PERMISSION_ADDRESS, StreamQueryResult, StreamRegistry } from './StreamRegistry'
+import { GraphQLClient } from './utils/GraphQLClient'
+import { filter, unique } from './utils/GeneratorUtils'
+
+export interface SearchStreamsPermissionFilter {
+    user: EthereumAddress
+    /*
+     * If possible, prefer allOf to anyOf because the query performance is better
+     */
+    allOf?: StreamPermission[]
+    anyOf?: StreamPermission[]
+    allowPublic: boolean
+}
+
+export type SearchStreamsQueryItem = {
```

Comment:
I find the name a bit confusing, isn't this part of the result set of the query? I thought this interface somehow relates to building the query. What about something like `SearchStreamsResultItem`?